packages/portfolio-deploy/src/contract-control.js

```diff
@@ -0,0 +1,203 @@
+  upgrade: M.callWhen(BundleIdShape).returns(VatUpgradeResultsShape),
+  terminate: M.callWhen(M.string()).returns(),
+  pruneChainStorage: M.callWhen(
+    M.recordOf(M.string(), M.arrayOf(M.string())),
+  ).returns(M.number()),
+});
+
+/**
+ * @template {ContractStartFunction} SF
+ * @typedef {StartedInstanceKit<SF> & {
+ *   label: string;
+ *   privateArgs: Parameters<SF>[1];
+ * }} UpgradeKit
+ */
+
+/**
+ * @param {Zone} zone
+ */
+export const prepareContractControl = zone => {
+  return zone.exoClass(
+    'ContractControl',
+    iface,
+    // TODO: promises can't be stored durably
+    /**
+     * @template {ContractStartFunction} SF
+     * @param {{
+     *   name: string,
+     *   agoricNamesAdmin: ERef<NameHubKit['nameAdmin']>,
+     *   board: ERef<Board>,
+     *   zoe: ERef<ZoeService>,
+     *   startUpgradable: ERef<StartUpgradable>,
+     *   orchestrationPowers: OrchestrationPowers & {
+     *     storageNode: Remote<StorageNode>;
+     *   },
+     *   kit?: (StartedInstanceKit<SF> & { privateArgs: Parameters<SF>[1] }),
+     * }} initial
+     */
+    initial => initial,
+    {
+      /** @param {string} bundleId */
+      async install(bundleId) {
+        trace('install', bundleId);
+        const { name, zoe, board, agoricNamesAdmin } = this.state;
+        const installation = await E(zoe).installBundleID(bundleId);
+        const installationAdmin =
+          E(agoricNamesAdmin).lookupAdmin('installation');
+        await E(installationAdmin).update(name, installation);
+        const boardId = await E(board).getId(installation);
+        trace({ name, bundleId, installation, boardId });
+        return { boardId, installation };
+      },
+
+      /**
+       * @template {ContractStartFunction} SF
+       * @param {object} opts
+       * @param {Installation<SF>} opts.installation
+       * @param {IssuerKeywordRecord} [opts.issuers]
+       * @param {CopyRecord} [opts.dataPrivateArgs]
+       */
+      async start({ installation, issuers, dataPrivateArgs }) {
+        const { name, startUpgradable, board, orchestrationPowers } =
+          this.state;
+        const boardId = E(board).getId(installation);
+        trace('startUpgradable', { installation, boardId });
+        const privateArgs = harden({
+          ...orchestrationPowers,
+          ...dataPrivateArgs,
+        });
+        const kit = await E(startUpgradable)({
+          label: name,
+          installation,
+          issuerKeywordRecord: issuers,
+          //   terms: customTerms,
+          privateArgs,
+        });
+        /** @type {UpgradeKit<SF>} */
+        const fullKit = harden({ ...kit, privateArgs });
+        this.state.kit = fullKit;
+
+        return fullKit;
+      },
+
+      /**
+       * @param {object} opts
+       * @param {string} opts.bundleId
+       * @param {IssuerKeywordRecord} [opts.issuers]
+       * @param {CopyRecord} [opts.dataPrivateArgs]
+       */
+      async deploy({ bundleId, issuers, dataPrivateArgs }) {
```

Comment:
Please document that "deploy" means "install and start", and consider just renaming it accordingly:
```suggestion
      async installAndStart({ bundleId, issuers, dataPrivateArgs }) {
```

Reply:
thanks for the suggestion; renamed.